knx/dpt/types_1.go

```diff
@@ -0,0 +1,556 @@
+
+func (d DPT_1008) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1008) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1008) Unit() string {
+	return ""
+}
+
+func (d DPT_1008) String() string {
+	if d {
+		return "Down"
+	} else {
+		return "Up"
+	}
+}
+
+// DPT_1009 represents DPT 1.009 (G) / DPT_OpenClose.
+type DPT_1009 bool
+
+func (d DPT_1009) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1009) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1009) Unit() string {
+	return ""
+}
+
+func (d DPT_1009) String() string {
+	if d {
+		return "Close"
+	} else {
+		return "Open"
+	}
+}
+
+// DPT_1010 represents DPT 1.010 (G) / DPT_Start.
+type DPT_1010 bool
+
+func (d DPT_1010) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1010) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1010) Unit() string {
+	return ""
+}
+
+func (d DPT_1010) String() string {
+	if d {
+		return "Start"
+	} else {
+		return "Stop"
+	}
+}
+
+// DPT_1011 represents DPT 1.011 (FB) / DPT_State.
+type DPT_1011 bool
+
+func (d DPT_1011) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1011) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1011) Unit() string {
+	return ""
+}
+
+func (d DPT_1011) String() string {
+	if d {
+		return "Active"
+	} else {
+		return "Inactive"
+	}
+}
+
+// DPT_1012 represents DPT 1.012 (FB) / DPT_Invert.
+type DPT_1012 bool
+
+func (d DPT_1012) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1012) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1012) Unit() string {
+	return ""
+}
+
+func (d DPT_1012) String() string {
+	if d {
+		return "Inverted"
+	} else {
+		return "Not inverted"
+	}
+}
+
+// DPT_1013 represents DPT 1.013 (FB) / DPT_DimSendStyle.
+type DPT_1013 bool
+
+func (d DPT_1013) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1013) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1013) Unit() string {
+	return ""
+}
+
+func (d DPT_1013) String() string {
+	if d {
+		return "Cyclically"
+	} else {
+		return "Start/stop"
+	}
+}
+
+// DPT_1014 represents DPT 1.014 (FB) / DPT_InputSource.
+type DPT_1014 bool
+
+func (d DPT_1014) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1014) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1014) Unit() string {
+	return ""
+}
+
+func (d DPT_1014) String() string {
+	if d {
+		return "Calculated"
+	} else {
+		return "Fixed"
+	}
+}
+
+// DPT_1015 represents DPT 1.015 (G) / DPT_Reset.
+type DPT_1015 bool
+
+func (d DPT_1015) Pack() []byte {
+	return packB1(bool(d))
+}
+
+func (d *DPT_1015) Unpack(data []byte) error {
+	return unpackB1(data, (*bool)(d))
+}
+
+func (d DPT_1015) Unit() string {
+	return ""
+}
+
+func (d DPT_1015) String() string {
+	if d {
+		return "reset command"
```

Comment:
Just a nit, but it would be nice of all the DPT descriptions would all be in the same case style (e.g. capitalised or lowercase).

Reply:
I adapted the descriptions to be as close as possible to what ETS5 is displaying, but I must confess that I have not checked all of them. Otherwise I tried to exactly adhere to what is written in Version 02.02.01 of the Datapoint Types Specification. There are still some typos in the specs (after 20 years, what would one expect?) like "Lux" and "lux" and a few others.

In that case, I shortened "reset command (trigger)" to "reset command".